src/future/future/select.rs

```diff
@@ -0,0 +1,53 @@
+use std::pin::Pin;
+
+use pin_project_lite::pin_project;
+use async_macros::MaybeDone;
+
+use std::future::Future;
+use crate::task::{Context, Poll};
+
+pin_project! {
+    #[allow(missing_docs)]
+    #[allow(missing_debug_implementations)]
+    pub struct Select<L, R> where L: Future, R: Future<Output = L::Output> {
```

Comment:
Are the bounds necessary here? I believe having them only on `impl` blocks should be enough.

Reply:
Just verified this, and yes they're needed. I think that's okay though; don't think we need to spend any time chasing this down (: